tests/unit/test_charm.py

```diff
@@ -987,3 +998,271 @@ def test_given_ca_cert_is_not_stored_when_certificate_transfer_relation_joins_th
         )
 
         self.assertNotIn("ca", relation_data)
+
+    @patch("vault.Vault.is_sealed", new=Mock)
+    @patch("vault.Vault.create_snapshot", new=Mock)
+    @patch("vault.Vault.unseal")
+    @patch("vault.Vault.is_initialized")
+    @patch("vault.Vault.is_api_available")
+    def test_given_vault_is_initialized_when_create_raft_snapshot_then_vault_is_unsealed(
+        self,
+        patch_is_api_available,
+        patch_is_initialized,
+        patch_unseal,
+    ):
+        patch_is_api_available.return_value = True
+        patch_is_initialized.return_value = True
+
+        self.harness.set_can_connect(container=self.container_name, val=True)
+        self.harness.add_storage(storage_name="certs", attach=True)
+        peer_relation_id = self._set_peer_relation()
+        self._set_initialization_secret_in_peer_relation(
+            relation_id=peer_relation_id,
+            root_token="root token content",
+            unseal_keys=["unseal_keys"],
+        )
+        self.harness.charm._create_raft_snapshot()
+        patch_unseal.assert_called_once()
+
+    @patch("vault.Vault.is_sealed", new=Mock)
+    @patch("vault.Vault.unseal", new=Mock)
+    @patch("vault.Vault.is_initialized")
+    @patch("vault.Vault.is_api_available")
+    @patch("vault.Vault.create_snapshot")
+    def test_given_vault_is_initialized_when_create_raft_snapshot_then_snapshot_is_created(
+        self,
+        patch_create_snapshot,
+        patch_is_api_available,
+        patch_is_initialized,
+    ):
+        patch_is_api_available.return_value = True
+        patch_is_initialized.return_value = True
+        self.harness.set_can_connect(container=self.container_name, val=True)
+        self.harness.add_storage(storage_name="certs", attach=True)
+        peer_relation_id = self._set_peer_relation()
+        self._set_initialization_secret_in_peer_relation(
+            relation_id=peer_relation_id,
+            root_token="root token content",
+            unseal_keys=["unseal_keys"],
+        )
+        self.harness.charm._create_raft_snapshot()
```

Comment:
We shouldn't test our private methods directly, here the public interface is really the Juju action handler (i.e. `_on_create_backup_action`) and this is what should be tested. This applies to all tests aimaing at `_create_raft_snapshot` directly.

Reply:
Updated